### ISSUE TITLE
[TimePicker] Can't change color of selected items

### ISSUE THREAD

**Issue by stesvis** [mui/mui-x#5194]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

In TimePicker I am unable to set the color of the selected items in the clock, while I was able to change basically everything else:
![image](https://user-images.githubusercontent.com/6062684/173454455-427b8279-7c03-4833-8480-899ba88e671f.png)

I would like to make the background color of the highlighted items red instead of blue (AM/PM and selected time)

### Expected behavior 🤔

The text color of the numbers should change.

### Steps to reproduce 🕹

I tried to accomplish this using the `PaperProps`
```
    <LocalizationProvider dateAdapter={AdapterDateFns}>
      <TimePicker
        disableMaskedInput
        //   Style Props --- start
        PaperProps={{
          sx: {
            backgroundColor: "#000000",
            color: "#eeeeee",
            "& .MuiButtonBase-root": {
              color: "#ffffff",
            },
            "& .MuiPickersArrowSwitcher-button": {
              backgroundColor: "#282d36",
              color: "#ffffff",
            },
            // clocl background color
            "& .css-eziifo": {
              backgroundColor: "#282d36",
            },
            // clock numbers color
            "& .css-1flhz3h": {
              color: "#ffffff",
            },
            // clock selected items color
            "& .css-7lip4c, .css-12ha4i7, .css-118whkv, .Mui-selected, .css-a1rc6s":
              {
                backgroundColor: "#d3121a",
              },
          },
        }}
        InputProps={{
          disableUnderline: true,
          sx: {
            backgroundColor: "#282d36",
          },
        }}
        OpenPickerButtonProps={{
          sx: { color: "hsl(0, 0%, 80%)" },
        }}
        //   Style Props --- end
        inputFormat="HH:mm"
        onChange={handleChange}
        renderInput={renderCustomInput}
        value={date}
      />
    </LocalizationProvider>
```

### Context 🔦

I am just trying to customize my UI elements to match the app theme I use.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Windows 10 10.0.19043
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD  
    npm: 8.12.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 102.0.5005.63
    Edge: Spartan (44.19041.1266.0), Chromium (102.0.1245.39)
  npmPackages:
    @emotion/react: ^11.9.0 => 11.9.0
    @emotion/styled: ^11.8.1 => 11.8.1
    @mui/base:  5.0.0-alpha.81
    @mui/icons-material: ^5.8.0 => 5.8.0
    @mui/material: ^5.8.0 => 5.8.0
    @mui/private-theming:  5.8.0
    @mui/styled-engine:  5.8.0
    @mui/styled-engine-sc: ^5.8.0 => 5.8.0
    @mui/system:  5.8.0
    @mui/types:  7.1.3
    @mui/utils:  5.8.0
    @mui/x-date-pickers: ^5.0.0-alpha.6 => 5.0.0-alpha.6
    @types/react:  18.0.8
    react: ^18.1.0 => 18.1.0
    react-dom: ^18.1.0 => 18.1.0
```
</details>

**Comment by alexfauquette** [comment]
As you probably guessed, classes `.css-eziifo` are hash that change at each render. To be able to do this customization, the Clock picker should be modified to add classnames `MuiPickersClock-[name of the component]`

The issue to track progress on this topic is here https://github.com/mui/mui-x/issues/4669#issuecomment-1115044196

The color can be modified with the theme, but it's clearly mode adapted to customize a single element
https://codesandbox.io/s/statictimepickerdemo-demo-mui-x-forked-dv7lb6?file=/demo.tsx

**Comment by flaviendelangle** [comment]
Closed by #6156
